ofctrl/transaction.go

```diff
@@ -121,23 +121,29 @@ func (tx *Transaction) listenReply() {
 	for {
 		select {
 		case reply, ok := <-tx.controlReplyCh:
-			if !ok { // controlReplyCh closed.
-				return
-			}
-			switch reply.msgType {
-			case BundleControlMessage:
-				select {
-				case tx.controlIntCh <- reply:
-				case <-time.After(messageTimeout):
-					log.Warningln("BundleControlMessage reply message accept timeout")
+			if func() bool {
+				tx.lock.Lock()
+				defer tx.lock.Unlock()
```

Comment:
Normally it's not good to hold the lock when their is a sleep. Could you add comments to explain what the lock protects from?